system/HTTP/ResponseTrait.php

```diff
@@ -503,28 +503,27 @@ public function sendBody()
      */
     public function redirect(string $uri, string $method = 'auto', ?int $code = null)
     {
-        // Assume 302 status code response; override if needed
-        if (empty($code)) {
-            $code = 302;
-        }
-
         // IIS environment likely? Use 'refresh' for better compatibility
         if (
             $method === 'auto'
             && isset($_SERVER['SERVER_SOFTWARE'])
             && strpos($_SERVER['SERVER_SOFTWARE'], 'Microsoft-IIS') !== false
         ) {
             $method = 'refresh';
+        } elseif ($method !== 'refresh' && $code === null) {
+            // override status code for HTTP/1.1 & higher
+            if (
+                isset($_SERVER['SERVER_PROTOCOL'], $_SERVER['REQUEST_METHOD'])
+                && $this->getProtocolVersion() >= 1.1
+            ) {
+                $code = ($_SERVER['REQUEST_METHOD'] !== 'GET')
+                    ? 303 // reference: https://en.wikipedia.org/wiki/Post/Redirect/Get
+                    : 307;
```

Comment:
For the HEAD and OPTIONS methods, there will be a 303 code, that is, a redirect with the GET method.

Reply:
Fixed.